spec/PeriodicBackgroundSync-index.bs

```diff
@@ -59,23 +70,222 @@ Introduction {#intro}
+    1. If |currentRegistration| is null:
+        1. Let |newRegistration| be a new [=periodicsync registration=].
+        1. Set |newRegistration|'s associated [=periodicsync registration/tag=] to |tag|.
+        1. Set |newRegistration|'s associated [=periodicsync registration/options=] to |options|.
+        1. Set |newRegistration|'s associated [=periodicsync registration/registration state=] to [=pending=].
+        1. Set |newRegistration|'s associated [=periodicsync registration/service worker registration=] to |serviceWorkerRegistration|.
+        1. Add |newRegistration| to |serviceWorkerRegistration|'s [=list of periodicsync registrations=].
+        1. [=Calculate a time to fire=] for |newRegistration|.
+        1. [=Schedule processing=] for |registration|.
+        1. [=Resolve=] |promise|.
+    1. Else:
+        1. If the |currentRegistration|'s [=periodicsync registration/options=] is different from |options|:
+            1. Set |currentRegistration|'s associated [=periodicsync registration/options=] to |options|.
+            1. [=Calculate a time to fire=] for |newRegistration|.
+            1. Set |currentRegistration|'s associated [=periodicsync registration/registration state=] to [=pending=].
+        1. Else, if |currentRegistration|'s [=periodicsync registration/registration state=] is [=firing=], set |serviceWorkerRegistration|'s [=periodicsync registration/registration state=] to [=reregisteredWhileFiring=].
+        1. [=Resolve=] |promise|.
+
+The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn></code> method when invoked, MUST return [=a new promise=] |promise| and  [=enqueue the following steps=] to the [=periodicsync processing queue=]:
+
+1. Let |serviceWorkerRegistration| be the {{PeriodicSyncManager}}'s associated [=/service worker registration=].
+1. Let |currentTags| be a new [=/list=].
+1. For each |registration| of |serviceWorkerRegistration|'s [=list of periodicsync registrations=], [=list/append=] |registration|'s [=periodicsync registration/tag=] to |currentTags|.
+1. [=Resolve=] |promise| with |currentTags|.
+
+The <code><dfn method for=PeriodicSyncManager title="unregister(tag)">unregister(tag)</dfn></code> method when invoked, MUST return [=a new promise=] <var>promise</var> and run the following steps:
+1. Let <var>serviceWorkerRegistration</var> be the {{PeriodicSyncManager}}'s associated [=/service worker registration=].
+1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+1. Else, [=enqueue the following steps=] to the [=periodicsync processing queue=]:
+    1. Let <var>currentRegistration</var> be the [=periodicsync registration=] in |serviceWorkerRegistration|'s [=list of periodicsync registrations=] whose [=periodicsync registration/tag=] equals <var>tag</var> if it exists, else null.
+    1. If |currentRegistration| is null:
```

Comment:
Rephrase as:

1. If `currentRegistration` is not null:
    1. Remove `registration` from `serviceWorkerRegistration`'s list of periodicsyncevents.
1. Resolve `promise`

Changes five steps into one.

Reply:
done.